org.osgi.test.cases.jdbc/src/org/osgi/test/cases/jdbc/junit/JDBCTestCase.java

```diff
@@ -77,20 +79,21 @@ private static class DfltProps extends Properties {
 	static String				description										= "desc";
 	static String				password										= "pswd";
 	static String				user											= "usr";
-
-	@Test
+	
+	/**
+	 * Test that the JDBS service defines at least one capability
+	 * 
+	 * @param sr the service reference parameter
+	 */
+	@ParameterizedTest
+	@ServiceSource(serviceType = DataSourceFactory.class)
```

Comment:
You have lost the filter expression `"(" + OSGI_JDBC_DRIVER_CLASS + "=*)"` from the test.

Reply:
This seems redundant here and is tested in `testRegisteredProperties` already.

Reply:
I think the point of this test method is to confirm there is at least one DSF with the driver class property and the capability. Which is why it was not a parameterized test. I am now thinking we should leave this test method alone. Making it parameterized does not add any value and seems to lose some of the purpose of the test.

Reply:
This makes no sense to me... `OSGI_JDBC_DRIVER_CLASS` is a required property, that means every service must implement it to be compliant. And only testing one here but multiple ones in the other methods seems really odd, so my implementation becomes compliant to the TCK if I register multiple ones where only some has the required properties?

Reply:
By the way does `@InjectService(cardinality = 1)` mean *exactly* one or *at laest* one? So it would be good to clarify if this should test exactly one service or several ones. If only one I still wonder why a filter is required and not the service type is enough?

And then the other methods should also simply collect one service instead of potentially multiple ones...

Reply:
The spec says:

> Each Data Source Factory service must relate to a single Database Driver. The Database Driver implementation bundle does not necessarily need to be the registrar of the Data Source Factory service. Any bundle can provide the Data Source Factory service and delegate to the appropriate driver specific implementation classes. 

This implies to me a compliant implementation can implement as many `DataSourceFactory`s as desired (it is unclear what "must relate to a single Database Driver" should mean in the context of the spec), so each of those factories must comply to any rules, and it is even invalid to check this in different methods as there is no guarantee the same number of services are checked.

So I think actually the checks that are now separate testing for properties are not compliant to the spec ... if you agree I'll therefore rework both PRs to take this into account.